src/app/InteractionModelEngine.h

```diff
@@ -238,6 +238,22 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
                       const PayloadHeader & aPayloadHeader, System::PacketBufferHandle && aPayload);
 
 #if CHIP_CONFIG_ENABLE_READ_CLIENT
+    /**
+     *  Activate the idle subscriptions.
+     *
+     *  When subscribing to ICD and liveness timeout reached, the read client will move to `InactiveICDSubscription` state and
+     * resubscription can be triggered via OnActiveModeNotification().
+     */
+    void OnActiveModeNotification(ScopedNodeId aPeer);
+
+    /**
+     *  Used to notify whether a peer becomes LIT ICD or vice versa.
+     *
+     *  When the app knows that the peer becoms LIT ICD, it is expected to call this method with PeerType::kLITICD and when
+     * the peer is nolonger a LIT ICD, it is expected to call this method with PeerType::kNormal.
+     */
+    void OnPeerTypeChange(ScopedNodeId aPeer, ReadClient::PeerType aType);
```

Comment:
Who would be able to call this, and on the basis of what information?

Reply:
How about to process OperatingMode attribute directly in readClient https://github.com/project-chip/connectedhomeip/blob/master/src/app/ReadClient.cpp#L737, then call OnPeerTypeChange? this way, the application, like chip-tool or ecosystem application don't need to handle operating mode attribute and expose/call OnPeerTypeChange

Reply:
thanks